airbyte-workers/src/main/java/io/airbyte/workers/temporal/TemporalClient.java

```diff
@@ -357,7 +357,33 @@ public ManualSyncSubmissionResult resetConnection(final UUID connectionId) {
       }
     } while (connectionManagerWorkflow.getJobInformation().getJobId() == oldJobId);
 
-    log.info("end of reset");
+    log.info("end of reset submission");
+
+    final long jobId = connectionManagerWorkflow.getJobInformation().getJobId();
+
+    return new ManualSyncSubmissionResult(
+        Optional.empty(),
+        Optional.of(jobId));
+  }
+
+  public ManualSyncSubmissionResult synchronousResetConnection(UUID connectionId) {
+    resetConnection(connectionId);
```

Comment:
Should this fail if a `ManualSyncSubmissionResult` is returned without a workflow running?

Reply:
Done